ibc-testkit/tests/core/ics02_client/create_client.rs

```diff
@@ -48,7 +49,7 @@ fn test_create_client_ok() {
 
     assert!(res.is_ok(), "execution happy path");
 
-    let expected_client_state = ctx.decode_client_state(msg.client_state).unwrap();
+    let expected_client_state = ClientStateRef::<MockContext>::try_from(msg.client_state).unwrap();
```

Comment:
Looks like now we have to remember the type of the context. Can we achieve something like this,

```rs
fn try_from_any<T>(_: &T, any_msg: Any) -> Result<ClientStateRef::<T>, ...> {
    ClientStateRef::<T>::try_from(any_msg)
}

let expected_client_state = try_from_any(msg.client_state).unwrap()
```

Reply:
I may not fully grasp your point, but it appears to be working efficiently as is.

Reply:
Ah sorry I wasn't clear. Here, you had to be explicit about `<MockContext>`. I was wondering if we can avoid it.

Note that `ctx` already has this type info. Currently, Rust doesn't have a way to extract type statically from a variable directly. My above suggestion is a working hack of the same.